csrc/ir/internal_nodes.h

```diff
@@ -2860,6 +2860,148 @@ class ArgsortOp : public Expr {
   }
 };
 
+//! NOTE -- [ Grouped Matrix Multiplication semantics ]
+//!
+//! This operation performs a grouped matrix multiplication.
+//!
+//! Parameters:
+//! - out: output tensor
+//! - matrix1: first input tensor matrix
+//! - matrix2: second input tensor matrix
+//! - offsets: 1D offsets tensor, specifying the ending index of each group
+//! - scale1: scale tensor for matrix1 (optional)
+//! - scale2: scale tensor for matrix2 (optional)
+//!
+//! The offsets tensor is a vector tensor of length `num_groups` that specifies
+//! the ending index of each group in the matrix1 and matrix2 tensors.
+//!
+//! Given the number of groups as G, the operation conceptually runs G matmuls.
+//! There are three configurations of grouping, reflected by ranks of input
+//! matrices:
+//!
+//! Note 0: f(i) = offset(i-1) : offset(i)
```

Comment:
So f(0) = input[offset(-1) : offset(0)]. Is offset(-1) == 0 and offset(g-1) == size(input)?

Reply:
I should have marked it explicitly `offset(-1) == 0`.

Meanwhile, I don't think it's required to have `offset(g-1)` to cover the entire inputs.
Technically, `offset` needs to be non-decreasing, but the kernel doesn't assert on that neither and we might get data racing.